src/rpc/core_rpc_server.cpp

```diff
@@ -717,8 +716,8 @@ namespace cryptonote
 
       if (might_be_blink)
       {
-        if (!pool_lock) pool_lock.lock();
-        e.blink = pool.has_blink(tx_hash, true /*have lock*/);
+        if (!blink_lock) blink_lock.lock();
```

Comment:
I think you can just declare and take the blink_lock here. Doesn't look like we do any more pool stuff with blink after this point that the lock needs to be held until scope exit.

Reply:
My main thinking here was to only take it out once per loop, but not to take it out at all if not needed (which will be pretty common when someone requests txs from old blocks rather than current pool txes)

Reply:
oh, yeah sorry, missed the loop.